src/wallet/db.h

```diff
@@ -119,10 +119,14 @@ class CWalletDBWrapper
      */
     void Flush(bool shutdown);
 
+    void IncrementUpdateCounter();
+    unsigned int GetUpdateCounter();
+
 private:
     /** BerkeleyDB specific */
     CDBEnv *env;
     std::string strFile;
+    std::atomic<unsigned int> nUpdateCounter;
```

Comment:
> It's an atomic, should have a default constructor?

Yes but it seems to not initialize the value, see
https://stackoverflow.com/questions/36320008/whats-the-default-value-for-a-stdatomic or http://en.cppreference.com/w/cpp/atomic/atomic/atomic. IMO, adding `= 0` to this line would make the code clearer anyway.